.build/build-git.xml

```diff
@@ -51,7 +51,34 @@
         <echo level="warning" if:true="${is-dirty}">Repository state is dirty</echo>
         <echo level="warning" if:true="${is-dirty}">${git.diffstat}</echo>
 
-        <property name="get-git-sha.done" value="true"/>
+        <exec if:true="${git.is-available}" executable="sh" osfamily="unix" dir="${basedir}" logError="true" failonerror="false" failifexecutionfails="false">
+            <arg value="-c"/>
+            <arg value="git remote -v | grep apache/cassandra | cut -f 1 | head -n 1"/>
+            <redirector outputproperty="_resolved_repo_name" description="Tries to find the remote name of apache/cassandra repository"/>
+        </exec>
+        <condition property="_ref_branch" value="${_resolved_repo_name}/trunk" else="trunk" description="This property contains 'apache/cassandra repo name'/trunk or just trunk if the repo name cannot be resolved">
+            <and>
+                <isset property="_resolved_repo_name"/>
+                <not><equals arg1="${_resolved_repo_name}" arg2=""/></not>
+            </and>
+        </condition>
+
+        <exec if:true="${git.is-available}" executable="git" osfamily="unix" dir="${basedir}" logError="true" failonerror="false" failifexecutionfails="false">
+            <arg value="diff"/>
+            <arg value="--name-only"/>
+            <arg value="--no-renames"/>
+            <arg value="--ignore-all-space"/>
+            <arg value="--merge-base"/>
+            <arg value="${_ref_branch}"/>
+            <arg value="${build.helpers.dir}/cassandra-build-deps-template.xml"/>
+            <arg value="${build.helpers.dir}/cassandra-deps-template.xml"/>
+            <arg value="${build.helpers.dir}/parent-pom-template.xml"/>
+            <redirector outputproperty="git.diff.deps-xmls"/>
+        </exec>
+        <filelist id="git.diff.deps-xmls.list" files="${git.diff.deps-xmls}" description="List of changed production sources between HEAD and the latest common ancestor of HEAD and trunk"/>
+        <pathconvert property="git.diff.deps-xmls.path" pathsep="," refid="git.diff.deps-xmls.list">
+        </pathconvert>
```

Comment:
why? can this go into a private ant target with a descriptive name pls.

Reply:
what exactly are you referring to?

Reply:
what does this addition of code/xml do? 

a private ant target with a descriptive name beats a comment.